hail/src/main/scala/is/hail/expr/ir/MatrixIR.scala

```diff
@@ -260,6 +269,9 @@ case class MatrixRangeReader(nRows: Int, nCols: Int, nPartitions: Option[Int]) e
     rowType = TStruct("row_idx" -> TInt32()),
     entryType = TStruct.empty())
 
+  def rvRowPType(requestedType: MatrixType): PStruct = PStruct("row_idx" -> PInt32(), MatrixType.entriesIdentifier -> PArray(PStruct()))
+  def physicalKey(requestedType: MatrixType): IndexedSeq[String] = requestedType.rowKey
```

Comment:
I'm going to suggest a slight re-org here:

 - MatrixIR, TableIR, etc. and MarixReader have RVDType instead of ptype + key
 - the RVDType of MatrixRead is the reader's RVDType restricted to the requested type

I feel like this will be a bit cleaner than what's forming.  Thoughts?

Reply:
yes, this is definitely better.